src/main/java/org/jenkinsci/plugins/DependencyCheck/DependencyCheckPublisher.java

```diff
@@ -107,40 +107,46 @@ public void perform(@Nonnull final Run<?, ?> build,
             build.setResult(Result.UNSTABLE);
             return;
         }
-        final ReportParser parser = new ReportParser(build.getNumber());
+        
+        final ReportParser parser = new ReportParser();
         for (FilePath odcReportFile: odcReportFiles) {
-            try {
-                final List<Finding> findings = parser.parse(odcReportFile.read());
-                final SeverityDistribution severityDistribution = parser.getSeverityDistribution();
-                final ResultAction projectAction = new ResultAction(build, findings, severityDistribution);
-                build.addAction(projectAction);
-
-                // Get previous results and evaluate to thresholds
-                final Run previousBuild = build.getPreviousBuild();
-                final RiskGate riskGate = new RiskGate(getThresholds());
-                if (previousBuild != null) {
-                    final ResultAction previousResults = previousBuild.getAction(ResultAction.class);
-                    if (previousResults != null) {
-                        final Result result = riskGate.evaluate(
-                                previousResults.getSeverityDistribution(),
-                                previousResults.getFindings(),
-                                severityDistribution,
-                                findings);
-                        evaluateRiskGates(build, logger, result);
-                    } else { // Resolves https://issues.jenkins-ci.org/browse/JENKINS-58387
-                        final Result result = riskGate.evaluate(severityDistribution, new ArrayList<>(), severityDistribution, findings);
-                        evaluateRiskGates(build, logger, result);
-                    }
-                } else { // Resolves https://issues.jenkins-ci.org/browse/JENKINS-58387
-                    final Result result = riskGate.evaluate(severityDistribution, new ArrayList<>(), severityDistribution, findings);
-                    evaluateRiskGates(build, logger, result);
-                }
-
-            } catch (InvocationTargetException | ReportParserException e) {
-                logger.log(Messages.Publisher_NotParsable() + " " + odcReportFile.getRemote());
-                logger.log(e.getMessage());
-            }
+        	try {
+        		logger.log(Messages.Publisher_ParsingFile() + " " + odcReportFile.getRemote());
+        		parser.parse(odcReportFile.read());
+        	} catch (InvocationTargetException | ReportParserException e) {
+        		logger.log(Messages.Publisher_NotParsable() + " " + odcReportFile.getRemote());
+        		logger.log(e.getMessage());
+        		build.setResult(Result.FAILURE);
+                return;
+        	}
         }
+	                
+        final SeverityDistribution severityDistribution = parser.getSeverityDistribution();
+        final List<Finding> findings = parser.getFindings();
```

Comment:
It would be cleaner to aggregate the findings in `DependencyCheckPublisher`. Parser just transforming the XML into Findings entity is a clearly defined responsibility and parser's state should not depend on previously parsed XMLs.

Reply:
I agree

Reply:
> Thanks for the fix! Any chances to merge it in the near future? With this bug it's impossible to create reasonable reports for mono-repos (Maven/Gradle with OWASP plugin + OWASP CLI for npm packages).

I don't understand. Why it is impossible for mono-repos. With the current state this also isn't working. Only the findings of the last found DependencyCheck report is shown if there are multiple files. That can't be the soulution for mono-repos.
If you have some detailed requirements for mono-repos, please tell me, and I will try to implement that.

Reply:
I just fixed a bug. You are free to refactor the code afterwards. I did not make your code more worse by fixing the bug this.

Reply:
The point is that parser should simply parse the XML report. The logic how to aggregate/merge strategy it's better  moved/implemented into a different class or leave merge strategy into the publisher

Reply:
I totaly agree on that, but that could be refactored later on.

Reply:
@mum-viadee This comment was on the current state of the plugin, not your change. With your fix everything should work as expected 👍

Reply:
@markiewiczart Sorry, I got that wrong.

Reply:
I added a separate class for the aggregation of the findings. This could be handy in future, because there could be other ways of aggregating the findings than just summarize them.